app/views/email/edit.html.erb

```diff
@@ -0,0 +1,17 @@
+<% content_for :page_title, 'Your email address' %>
+<%= content_for :back_link_url, controller.request.env['HTTP_REFERER'].include?('check-answers') ? check_answers_path : start_path %>
```

Comment:
Yeah... I've been down this path...

This won't work if the user refreshes the webpage; the referrer will get lost. And the back path will reset to the start page, which isn't the page they came from...

There's an alternative solution using `history.back` from the browser, but that doesn't work without JavaScript, and can also be fooled sometimes.

I don't think it's avoidable, but we need to somehow keep track of a bit of state which can be used to answer "Has this user reached the check your answers page? If yes, take them back there. If not, take them to the previous step in the form."

So we'd probably need a boolean that gets set to `true` when we reach the check your answers page, unless we can infer this state based on other information on the `TrnRequest` object.

cc @fofr I know you've probably spent a while dealing with / thinking about this problem.

Reply:
How about we use the referrer when it exists and when it is absent, we check the state of the `TrnRequest#email` field and use its presence as an indicator that the check answers screen should be the target of the back link?

Reply:
Hmm, I think that could work. 👍